### PR TITLE
feat: add windows testing tf + python

### PR DESCRIPTION
**Pull request by obs-gh-nikhildua**
**Description:**

Addresses: https://observe.atlassian.net/browse/CON-1395 

- Updates AWS EC2 only (Azure/GCP will be different ticket) to include Windows Server 2016, 2019, 2022
- Updates EC2 to include / decode / output password data for Windows  (passwords are sensitive and masked in python test code as well)
- Updates AWS Stack to include `networking.tf` allowing creation of vpc, subnet, gateway, route tables (to prevent relying on data sources) --> Thunderdome will kill resources so this is necessary 
- Updates python test scripts to support windows testing
- Adds powershell startup scripts (tested across all servers) to enable PS / SSH control 

**Testing**:
- See GH Actions run as part of this PR  and outcome (https://github.com/observeinc/linux-host-configuration-scripts/actions/runs/5765405161)  
- Extensive TF + Local Python testing 
- Note to test windows_branch in https://github.com/observeinc/windows-host-configuration-scripts, feed in the `windows_repo_branch` var and it'll use that branch to generate the appropriate windows curl command

Screenshot 2023-08-04 at 1 50 13 PM

### CODE REVIEW

File: test_code/AWS_MACHINES/data.tf [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
-}
 
-data "aws_subnet" "main" {
-  vpc_id = data.aws_vpc.main.id
-  filter {
-    name   = "tag:Name"
-    values = ["linux-host-test-subnet"]
-  }
-}
+### As of 08/03, commented out in favor of networking.tf ##
```
Comment (yasar-observe) [inline test_code/AWS_MACHINES/data.tf:7]: nit: If we don't need this, why not just remove it altogether?  If someone needs it in the future, we have it in git to reference.

**Review by yasar-observe — APPROVED** [review]

File: test_code/AWS_MACHINES/data.tf (hunk 1 above)
Comment (obs-gh-nikhildua) [reply test_code/AWS_MACHINES/data.tf:7]: Good point. Let me remove that

**Comment by obs-gh-nikhildua** [comment]
Ignore errors related to actions associated to PR. Running into edge case where overflow condition on `mkdir context` is hit if PR is opened AND reviewers are requested. (This condition is not hit if PR is opened and no reviewers added yet or in draft mod)

**Comment by obs-gh-nikhildua** [comment]
Updated Actions with latest commit: https://github.com/observeinc/linux-host-configuration-scripts/actions/runs/5767234402

**Pull request merged**